packages/scoutgame/src/points/getPartnerRewards.ts

```diff
@@ -0,0 +1,121 @@
+import { prisma } from '@charmverse/core/prisma-client';
+import type { Season } from '@packages/dates/config';
+import { getCurrentSeasonStart, getSeasonWeekFromISOWeek } from '@packages/dates/utils';
+
+type PartnerRewardBase = {
+  week: number;
+  points: number;
+  type: 'optimism_new_scout_partner' | 'optimism_top_referrer';
+  season: Season;
+};
+
+export type OptimismNewScoutPartnerReward = PartnerRewardBase & {
+  type: 'optimism_new_scout_partner';
+  partner: string;
+  position: number;
+};
+
+export type OptimismTopReferrerReward = PartnerRewardBase & {
+  type: 'optimism_top_referrer';
+  partner: string;
+  date: string;
+};
+
+export type PartnerReward = OptimismNewScoutPartnerReward | OptimismTopReferrerReward;
+
+export async function getUnclaimedPartnerRewards({ userId }: { userId: string }) {
+  const partnerRewards = await prisma.partnerRewardPayout.findMany({
+    where: {
+      payoutContract: {
+        season: getCurrentSeasonStart()
+      },
+      userId,
+      claimedAt: null
+    },
+    select: {
+      id: true,
+      amount: true,
+      payoutContract: {
+        select: {
+          partner: true
+        }
+      }
+    }
+  });
+
+  return partnerRewards.map((payout) => ({
+    id: payout.id,
+    amount: payout.amount,
```

Comment:
remember we need to convert the amount to something that's user-friendly. it's stored in gwei right, so at least convert using the correct decimals.